flink-yarn/src/test/java/org/apache/flink/yarn/YarnApplicationMasterRunnerTest.java

```diff
@@ -108,5 +131,39 @@ public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
 		ContainerLaunchContext ctx = Utils.createTaskExecutorContext(flinkConf, yarnConf, env, tmParams,
 			taskManagerConf, workingDirectory, taskManagerMainClass, LOG);
 		assertEquals("file", ctx.getLocalResources().get("flink.jar").getResource().getScheme());
+
+		Credentials credentials = new Credentials();
+		try (DataInputStream dis = new DataInputStream(new ByteArrayInputStream(ctx.getTokens().array()))) {
+			credentials.readTokenStorageStream(dis);
+		}
+		Collection<Token<? extends TokenIdentifier>> tokens = credentials.getAllTokens();
+		boolean hasHdfsDelegationToken = false;
+		boolean hasAmRmToken = false;
+		for (Token<? extends TokenIdentifier> token : tokens) {
+			if (token.getKind().equals(amRmTokenKind)) {
+				hasAmRmToken = true;
+			} else if (token.getKind().equals(hdfsDelegationTokenKind)) {
+				hasHdfsDelegationToken = true;
+			}
+		}
+		assertTrue(hasHdfsDelegationToken);
+		assertFalse(hasAmRmToken);
+	}
+
+	// A helper method used for setting system env variables for the test.
+	// See https://stackoverflow.com/a/496849/6558955
+	private static void setEnv(Map<String, String> newEnv) throws Exception {
```

Comment:
Maybe you could use `org.apache.flink.core.testutils.CommonTestUtils#setEnv`

Reply:
@klion26 Thanks for the pointer! I'll try it.